content/dev_stories/playground_to_production_fruit_falling.md

```diff
@@ -0,0 +1,677 @@
+    vulnerable to variations in framerate; so, borrowing from physics 
+    simulation practices, my solution was to make a simple
+    [fixed framerate observable](https://github.com/syntheticmagus/fruit-falling-source/blob/main/app_package/src/fixedFramerateObservable.ts)
+    which drives a guaranteed number of "updates" per second. If the device
+    is rendering faster than the desired framerate, this observable will
+    "skip" frames until enough time has elapsed; and if the device is 
+    rendering slower than the desired framerate, this observable will 
+    "double up" on what frames it has so that the same number of "updates"
+    per time are always called. This trick allowed me to write logic that
+    was *frame*-dependent without it becoming vulnerably 
+    frame*rate*-dependent.
+
+    ![Frame-dependent game logic](/img/devVignettes/fruitFalling/20_frame_dependent_logic.jpg!500)
+
+    Having this ability to reliably use frame-based logic allowed me to use
+    coroutines for all sorts of things, from game flow to procedural 
+    animation. A full overview of the use of coroutines in *Fruit Fallin'*
+    is beyond the scope of this article, but perhaps the most fun example 
+    look at is how 
+    [the face animations are done entirely using coroutines](https://github.com/syntheticmagus/fruit-falling-source/blob/main/app_package/src/rainbowButton.ts#L102-L183).
+1.  I made the consuming app (in this repo, the test app) responsible for 
+    specifying the locations of all the assets used by the game.
+
+    As discussed in the Babylon docs on the 
+    [Template Repository Workflow](https://doc.babylonjs.com/toolsAndResources/templateRepositories#the-template-repository-workflow),
+    one of the most varied challenges in deploying Babylon experiences is
+    distributing assets. It's best to keep this problem completely separate
+    from actually *using* the assets within the code, thus allowing the 
+    same experience code to be used in multiple situations even when assets
+    are distributed differently. For this reason, my rule for myself is that
+    there should never be a hard-coded asset URL anywhere in `app_package`.
+    Instead, all resource locations should be passed in by the consuming
+    experience, whether that be `test_package` or some production shipping
+    vehicle. That way the `app_package` code can be written to "just work"
+    as long as it's given the right asset locations, and all any consuming
+    experience has to do to integrate it is just tell `app_package`
+    the correct URLs to use.
+    
+    ![index.js asset location code](/img/devVignettes/fruitFalling/21_asset_location_code.jpg!500)
+
+I should also briefly mention how I created the assets for the game. 
+Again, this is not an implementation overview; but for those who are 
+curious, here are the various tools I used to make *Fruit Fallin'*.
+
+1.  **Art:** I drew all the backgrounds, the buttons, and all the 

```

Comment:
Pretty cool section, even shows how simple is to create assets! Linking https://opengameart.org/ as an option here could be cool too.

Reply:
I didn't use that here as I created all the assets myself, but we should definitely find a way to work that into a future Story and call it out there!